crates/turborepo-paths/src/anchored_system_path.rs

```diff
@@ -0,0 +1,84 @@
+use std::{
+    borrow::Cow,
+    fmt,
+    path::{Component, Path},
+};
+
+use path_slash::CowExt;
+
+use crate::{AnchoredSystemPathBuf, PathError};
+
+pub struct AnchoredSystemPath(Path);
+
+impl ToOwned for AnchoredSystemPath {
+    type Owned = AnchoredSystemPathBuf;
+
+    fn to_owned(&self) -> Self::Owned {
+        AnchoredSystemPathBuf(self.0.to_owned())
+    }
+}
+
+impl AsRef<AnchoredSystemPath> for AnchoredSystemPath {
+    fn as_ref(&self) -> &AnchoredSystemPath {
+        self
+    }
+}
+
+impl fmt::Display for AnchoredSystemPath {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.display().fmt(f)
+    }
+}
+
+impl AsRef<Path> for AnchoredSystemPath {
+    fn as_ref(&self) -> &Path {
+        &self.0
+    }
+}
+
+impl AnchoredSystemPath {
+    pub unsafe fn new_unchecked<'a>(path: impl AsRef<Path> + 'a) -> &'a Self {
+        let path = path.as_ref();
+        unsafe { &*(path as *const Path as *const Self) }
+    }
+
+    pub fn new<'a, T: AsRef<Path> + 'a>(path: T) -> Result<&'a Self, PathError> {
+        let path_ref = path.as_ref();
+        if path_ref.is_absolute() {
+            return Err(PathError::NotRelative(
+                path_ref.to_string_lossy().to_string(),
+            ));
+        }
+
+        let path_str = path_ref
+            .to_str()
+            .ok_or_else(|| PathError::InvalidUnicode(path_ref.to_string_lossy().to_string()))?;
+
+        let system_path = Cow::from_slash(path_str);
+        if let Cow::Owned(path) = system_path {
+            Err(PathError::NotSystem(path.to_string_lossy().to_string()))
+        } else {
+            Ok(unsafe { &*(path_ref as *const Path as *const Self) })
+        }
```

Comment:
Slight nit on this check is a noop on non-Windows and and on Windows can be simplified to a simple string check
```suggestion
        #[cfg(windows)]
        if path_str.contains('/') {
            return Err(PathError::NotSystem(path.to_string_lossy().to_string()))
        }
        Ok(unsafe { &*(path_ref as *const Path as *const Self) })
```

Reply:
Yeah the UTF-8 PR overhauls this entire setup and removes path-slash as a dependency